src/dialog/alert.js

```diff
@@ -13,28 +13,44 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2015-2019 (original work) Open Assessment Technologies SA ;
+ * Copyright (c) 2015-2021 (original work) Open Assessment Technologies SA ;
  */
 /**
  * @author Jean-Sébastien Conan <jean-sebastien.conan@vesperiagroup.com>
  */
 import _ from 'lodash';
+import __ from 'i18n';
 import dialog from 'ui/dialog';
 
 /**
- * Displays an alert message
- * @param {String} message - The displayed message
- * @param {Function} action - An action called when the alert is closed
- * @param {Function} onCreateDialog - An action called when dialog created
- * @returns {dialog} - Returns the dialog instance
- */
-export default function dialogAlert(message, action, onCreateDialog) {
-    var dlg = dialog({
+  * Displays an alert message
+  * @param {String} message - The displayed message
+  * @param {Function} action - An action called when the alert is closed
+  * @param {Function} onCreateDialog - An action called when dialog created
+  * @param {Object} options - Dialog options
+  * @param {Object} options.buttons - Dialog button options
+  * @param {Object} options.buttons.labels - Dialog button labels
+  * @param {String} options.buttons.labels.ok - "OK" button label
+  * @returns {dialog} - Returns the dialog instance
+  */
+export default function dialogAlert(message, action, onCreateDialog, options) {
+    const _options = {
+        buttons: {
+            labels: {
+                ok: __('Ok')
+            }
+        }
+    };
+    let dialogOptions;
+    let dlg;
+    options = _.defaults(options || {}, _options);
+    dialogOptions = {
         message: message,
-        buttons: 'ok',
+        buttons: options.buttons.labels.ok || __('Ok'),
```

Comment:
**nitpick:** This won't work as expected. When the button is supplied as a string, it is considered as an identifier, that will be use to link the built-in buttons. If the supplied id does not match a built-in button, it will generate a custom one, the label being the taken from the id. This might be ok if: the `close` flag was not missing (without it the dialog won't autoclose), and if the id was not taken from a translated label that will introduce regression when the id will be expected.

See: https://github.com/oat-sa/tao-core-ui-fe/blob/fbb967c48b8c68b1172a4b59c5b1d37e3933788e/src/dialog.js#L209-L222

**suggestion:** look at the implementation of the confirm dialog for a better approach: https://github.com/oat-sa/tao-core-ui-fe/blob/c11cec547adb9924adebd3d0a965c6c648e91e0e/src/dialog/confirm.js#L39-L74

Reply:
Thank you @jsconan.
I just test the label and click only in modern browsers.
But we should be careful here, I did a kind of port of confirm dialog, and then I had a security issue with fullscreen,
because of onOkBtn: fn()... is executed through `call`, it was blocked:
<img width="1904" alt="err" src="https://user-images.githubusercontent.com/2067027/112609752-34bd9400-8e1c-11eb-8d0d-7ae34c6f6d8a.png">